src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/ObjectPools/ArrayBuilder.cs

```diff
@@ -6,11 +6,10 @@ namespace Microsoft.CodeAnalysis.PooledObjects;
 
 internal sealed partial class ArrayBuilder<T> : IPooled
 {
+    private static readonly ObjectPool<ArrayBuilder<T>> s_keepLargeInstancesPool = CreatePool();
+
     public static PooledDisposer<ArrayBuilder<T>> GetInstance(out ArrayBuilder<T> instance)
-    {
-        instance = GetInstance();
-        return new PooledDisposer<ArrayBuilder<T>>(instance);
-    }
+        => GetInstance(discardLargeInstances: true, out instance);
```

Comment:
>discardLargeInstances [](http://example.com/codeflow?start=23&length=21)

I like the array builder change, just wondering if it might make sense to have it in it's own PR so others interested in that change wouldn't need to look at the nav search stuff.

Also, feels like this might have conflicts with Tomas's PR: [https://github.com/dotnet/roslyn/pull/76971](https://github.com/dotnet/roslyn/pull/76971)

Reply:
extracted out to just this.